### PR TITLE
[Remove] Type mapping parameter from put index mappings API

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description
With types deprecation, this change removes type mapping API end-points for put index mappings API end-point.

Relates https://github.com/opensearch-project/OpenSearch/issues/1940
 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
Needs rebase against main once https://github.com/opensearch-project/OpenSearch/pull/2220 is merged

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_mapping/20_mix_typeless_typeful.yml [hunk 1]
```diff
@@ -41,7 +38,6 @@
  - do:
       catch: /the final mapping would have more than 1 type/
       indices.put_mapping:
-          include_type_name: true
           index: index
           type: some_other_type
```
Comment (reta) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_mapping/20_mix_typeless_typeful.yml:42]: The presence of `type` seems to be not right, it should not be used, right?

Comment (dreamer-89) [reply rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_mapping/20_mix_typeless_typeful.yml:42]: @reta : Thanks for pointing this out. This also explains the existing test failures.

**Comment by dreamer-89** [comment]
Removal of `type` from rest end-points cause failures in yaml rest api tests because of validation checks on non-empty type in PutMappingRequest class. This mandates either adding back a dummy `_doc` type in put APIs or remove all internal references of type (and validations) in internal classes.

**Comment by dreamer-89** [comment]
Resolved in https://github.com/opensearch-project/OpenSearch/pull/2238

**Pull request closed without merging**